### PR TITLE
[WIP] Fix the "$elemmatch needs an object" exception

### PR DESCRIPTION
**Pull request by coudenysj**
This follows #1349 to fix the ``$elemmatch needs an object`` exception

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/Query/Expr.php [hunk 1]
```diff
@@ -106,12 +112,18 @@ public function includesReferenceTo($document)
         if ($this->currentField) {
             $mapping = $this->getReferenceMapping();
             $dbRef = $this->dm->createDBRef($document, $mapping);
+            $storeAs = array_key_exists('storeAs', $mapping) ? $mapping['storeAs'] : null;
 
-            if (isset($mapping['simple']) && $mapping['simple']) {
-                $this->query[$mapping['name']]['$elemMatch'] = $dbRef;
+            if ($storeAs === ClassMetadataInfo::REFERENCE_STORE_AS_ID) {
+                // TODO: This creates an invalid query ($elemMatch needs an Object)
```
Comment (coudenysj) [inline lib/Doctrine/ODM/MongoDB/Query/Expr.php]: Leaving a line note to make sure this TODO has been resolved before merging this pull request.

File: tests/Doctrine/ODM/MongoDB/Tests/Query/BuilderTest.php [hunk 2]
```diff
@@ -77,7 +89,25 @@ public function testIncludesReferenceToGoesThroughDiscriminatorMap()
             ->field('featureSimpleMany')->includesReferenceTo($f)
             ->getQuery()->debug();
 
+        // TODO: Looks like this query is not valid ($elemMatch needs an Object)
```
Comment (coudenysj) [inline tests/Doctrine/ODM/MongoDB/Tests/Query/BuilderTest.php]: Leaving a line note to make sure this TODO has been resolved before merging this pull request.

**Comment by alcaeus** [comment]
One thing about this: the fix needs to target the 1.0 branch since it's already an issue in 1.0.x and should be fixed there as well. So, I'd recommend creating a new PR based on the 1.0.x branch containing only the fix. Once that is merged (and downmerged into master) we can merge #1349.

**Comment by coudenysj** [comment]
This pull request has been replaced with #1406 and once that one has been merged in 1.0.x, it will need to be merged into master so I can rebase #1349.

**Pull request closed without merging**